src/cpp/lua-module.cpp

```diff
@@ -34,41 +25,94 @@ sol::object createChannel(const sol::stack_object& capacity, sol::this_state lua
 
 SharedTable globalTable = GC::instance().create<SharedTable>();
 
-std::string getLuaTypename(const sol::stack_object& obj)
-{
+std::string getLuaTypename(const sol::stack_object& obj) {
     return luaTypename<>(obj);
 }
 
+size_t luaSize(const sol::stack_object& obj) {
+    if (obj.is<SharedTable>())
+        return SharedTable::luaSize(obj);
+    else if (obj.is<Channel>())
+        return obj.as<Channel>().size();
+
+    throw effil::Exception() << "Unsupported type "
+                             << luaTypename(obj) << " for effil.size()";
+}
+
+sol::table luaThreadConfig(sol::this_state state, const sol::stack_object& obj) {
```

Comment:
Maybe it would be more readable to write these functions in lua code?

```
lua.script(R"(
api.thread = function (f)
    if type(f) ~= "function" then
        error("bad argument #1 to 'effil.thread' (function expected, got " .. capi.type(f) .. ")")
    end

    local thread_config = {
        path = package.path,
        cpath = package.cpath,
        step = 200 }
    setmetatable(thread_config, {__call = function(c, ...) return run_thread(c, f, ...) end})
    return thread_config
end
)");
```

Reply:
considering that it's already on c++ side, it's better to write it in c++. it's easier to write (cause Lua script will be just a huge string) and easy to debug, and we can imagine that it would be faster.